### PR TITLE
Gate Visualization

### PR DESCRIPTION
**Pull request by Steppschuh**
device-2019-08-19-135733

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -25,7 +25,8 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-    implementation 'com.nexenio.seamlessauthentication:core:0.8.0'
+    //implementation 'com.nexenio.seamlessauthentication:core:0.8.0'
```
Comment (Steppschuh) [inline app/build.gradle]: Use release version

File: build.gradle [hunk 2]
```diff
@@ -15,6 +15,14 @@ buildscript {
 
 allprojects {
     repositories {
+        // TODO: remove local artifactory
+        maven {
```
Comment (Steppschuh) [inline build.gradle]: Remove local artifactory

File: app/src/main/java/com/nexenio/seamlessauthenticationintegrationsample/visualization/GateView.java [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+    }
+
+    private void drawGatewayOpening(Canvas canvas, GatewayOpening gatewayOpening) {
+        int saveCount = canvas.save();
+
+        int direction = gatewayOpening.getDirection().blockingGet();
+        int lockMode = gate.getDirectionLockMode().blockingGet();
+        boolean isEntry = direction == GatewayDirection.ENTRY;
+        double distance = gatewayOpening.getDistance().onErrorReturnItem(99.0).blockingGet();
+        boolean isAllowed = GatewayMode.modeAllowsDirection(lockMode, direction);
```
Comment (Steppschuh) [inline app/src/main/java/com/nexenio/seamlessauthenticationintegrationsample/visualization/GateView.java:394]: Avoid allocations in `onDraw` and `onMeasure`

File: app/src/main/java/com/nexenio/seamlessauthenticationintegrationsample/visualization/GateView.java [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+
+        canvas.restoreToCount(saveCount);
+    }
+
+    private void drawBeaconAdvertisingPacket(Canvas canvas, AdvertisingPacket advertisingPacket) {
+        /*
+        long maximumTimestamp = System.currentTimeMillis() - 250;
+        long minimumTimestamp = maximumTimestamp - TimeUnit.SECONDS.toMillis(2);
+        float recencyScore = getRecencyScore(advertisingPacket.getTimestamp(), minimumTimestamp, maximumTimestamp);
+        */
```
Comment (Steppschuh) [inline app/src/main/java/com/nexenio/seamlessauthenticationintegrationsample/visualization/GateView.java]: Implement drawing of advertising packets

**Review by marvinmirtschin — APPROVED** [review]

**Pull request merged**